sdk/core/Azure.Core/tests/TestFramework/TestEnvironment.Recorded.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Identity;
+
+namespace Azure.Core.Testing
+{
+    public partial class TestEnvironment
+    {
+
+        private TestRecording _recording;
+        private bool _playback;
+
+        public void SetRecording(TestRecording recording, bool playback)
+        {
+            _recording = recording;
+            _playback = playback;
+        }
+
+        partial void GetRecordedValue(string name, ref string value, ref bool isPlayback)
+        {
+            if (_recording == null)
+            {
+                return;
+            }
+
+            value =  _recording.GetVariable(name, null);
+            isPlayback = _playback;
+        }
+
+        partial void SetRecordedValue(string name, string value)
+        {
+            _recording?.SetVariable(name, value);
+        }
+
+        public TokenCredential Credential => _playback ? (TokenCredential)new TestCredential() : new ClientSecretCredential(
```

Comment:
Can we make this a method?

Reply:
We can but why?

Reply:
Because it creates new instance of the object for every call.

Reply:
I can cache it if that's the concern.